m2cgen/interpreters/interpreter.py

```diff
@@ -228,3 +228,51 @@ def _cache_reused_expr(self, expr, expr_result):
         self._cached_expr_results[expr] = CachedResult(
             var_name=var_name, expr_result=None)
         return var_name
+
+
+class FunctionalToCodeInterpreter(ToCodeInterpreter):
```

Comment:
`interpret_pow_expr` and `interpret_log1p_expr` also seem to have the exact same implementations for F# and Haskell

Reply:
I don't think that interpreters of particular functions should be in the base class. I believe it should be as much general as possible. There are a lot more functional programming languages than F# and Haskell. For example, Clojure and Erlang have `pow` function.